spec/requests/conversations/update_spec.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'PUT /conversations/:id', type: :request do
+  let(:conversation) { create :conversation_with_messages, unread: true }
+  let(:params) do
+    {
+      conversation: {
+        unread: unread
+      }
+    }
+  end
+  let(:user) { conversation.initiator }
+
+  subject do
+    put "/conversations/#{conversation.id}", params: params, headers: headers
+    data
+  end
+
+  context "given I'm the second last user to send a message in the conversation" do
+    before :each do
+      create :message, user: conversation.target.user, conversation: conversation
+    end
+
+    context 'and I want to mark the conversation as read' do
+      let(:unread) { false }
+
+      it 'updates the conversation' do
+        expect { subject }.to change { conversation.reload.unread }.from(true).to(false)
+      end
+    end
+
+    context 'and the conversation is already read' do
+      let(:unread) { true }
+
+      before { conversation.update(unread: false) }
+
+      it 'updates the conversation to be marked as unread' do
+        expect { subject }.to change { conversation.reload.unread }.from(false).to(true)
+      end
+    end
+  end
+
+  context "given I'm the very last user to send a message in the conversation" do
```

Comment:
Try to make the contexts not in first person:

context `given the user is the last one to send a message in the conversation`